Make specs compatible with jquery v3

We can't currently update the jquery dependency used in our tests because 6 tests are incompatible with jquery3.  Again this isn't Marionette's code, that works fine, but the tests themselves are using deprecated functionality.

```
1) composite view when rendering with a false template should bind the ui hash to jQuery selectors:
     AssertionError: expected undefined to equal '.test-element'
      at Context.<anonymous> (composite-view.spec.js:950:61)

  2) region when initializing a region and passing an "el" option should manage the specified el:
     AssertionError: expected undefined to equal '#foo'
      at Context.<anonymous> (region.spec.js:807:43)

  3) layoutView when defining region selectors using @ui. syntax should apply the relevant @ui. syntax selector to a simple string value:
     AssertionError: expected undefined to equal '.craft'
      at Context.<anonymous> (view.child-views.spec.js:558:64)

  4) layoutView when defining region selectors using @ui. syntax should apply the relevant @ui. syntax selector to selector in a region definition object:
     AssertionError: expected undefined to equal '.bros'
      at Context.<anonymous> (view.child-views.spec.js:562:66)

  5) layoutView when defining region selectors using @ui. syntax should apply the relevant @ui. syntax selector to el in a region definition object:
     AssertionError: expected undefined to equal '.toadstool'
      at Context.<anonymous> (view.child-views.spec.js:566:69)

  6) item view when rendering with a false template should bind the ui hash to jQuery selectors:
     AssertionError: expected undefined to equal '.test-element'
      at Context.<anonymous> (view.spec.js:89:52)
```

>2) region when initializing a region and passing an "el" option should manage the specified el:
AssertionError: expected undefined to equal '#foo' at Context.<anonymous> (region.spec.js:807:43)

This is the only test that you can work around by doing `expect(this.region.el).to.equal(this.el);`

The other test can't use `.matches` because ui values are empty jQuery instances and you can't match against anything in them. Either the test go away since they are trying to test validity through a removed property from a 3rd party library, or the test have to move and test for elements that exist as DOM and match the selector test (not to be confused with `.selector`).

So... should I
1) remove these deprecated test 
or 
2) replace test with DOM match test somewhere else where the DOM elements do exist?

--- a/test/unit/composite-view.spec.js
+++ b/test/unit/composite-view.spec.js
@@ -946,10 +946,6 @@ describe('composite view', function() {
       expect(this.bindUIElementsSpy).to.have.been.calledOnce;
     });
 
-    it('should bind the ui hash to jQuery selectors', function() {
-      expect(this.compositeView.ui.testElement.selector).to.equal('.test-element');
-    });
-
     it('should trigger a before:render event', function() {
       expect(this.triggerSpy).to.have.been.calledWith('before:render', this.compositeView);
     });

--- a/test/unit/region.spec.js
+++ b/test/unit/region.spec.js
@@ -804,7 +804,7 @@ describe('region', function() {
     });
 
     it('should manage the specified el', function() {
-      expect(this.region.$el.selector).to.equal(this.el);
+      expect(this.region.el).to.equal(this.el);
     });
   });
 

--- a/test/unit/view.child-views.spec.js
+++ b/test/unit/view.child-views.spec.js
@@ -555,15 +555,12 @@ describe('layoutView', function() {
 
     it('should apply the relevant @ui. syntax selector to a simple string value', function() {
       expect(this.layoutView.getRegion('war')).to.exist;
-      expect(this.layoutView.getRegion('war').$el.selector).to.equal('.craft');
     });
     it('should apply the relevant @ui. syntax selector to selector in a region definition object', function() {
       expect(this.layoutView.getRegion('mario')).to.exist;
-      expect(this.layoutView.getRegion('mario').$el.selector).to.equal('.bros');
     });
     it('should apply the relevant @ui. syntax selector to el in a region definition object', function() {
       expect(this.layoutView.getRegion('princess')).to.exist;
-      expect(this.layoutView.getRegion('princess').$el.selector).to.equal('.toadstool');
     });
   });
 

--- a/test/unit/view.spec.js
+++ b/test/unit/view.spec.js
@@ -85,10 +85,6 @@ describe('item view', function() {
       expect(this.bindUIElementsSpy).to.have.been.calledOnce;
     });
 
-    it('should bind the ui hash to jQuery selectors', function() {
-      expect(this.view.ui.testElement.selector).to.equal('.test-element');
-    });
-
     it('should trigger a before:render event', function() {
       expect(this.triggerSpy).to.have.been.calledWith('before:render', this.view);
     });

